///**************************************************************************
//* @Project : cdzchx_camctrlserver
//* @IDL Module: ZCHXFllow
//* @File: ZCHXFllow.proto
//* @Description:
//*    
//*
//* @Copyright: Copyright (c) 2017
//* @Company: 深圳置辰海信科技有限公司
//* @WebSite: http://www.szcenterstar.com/
//* @author 
//*
//* @Revision History
//*
//* <pre>
//* ----------------------------------------------------------------------
//*   Ver     Date       Who             Comments
//*  ----- ----------  --------  ---------------------------------------
//*   1.0  2017/02/07     邓华钢       初次定义完成Com口的数据结构
//* ----------------------------------------------------------------------
//* </pre>
//**************************************************************************/


package com.zhichenhaixin.proto;
	message		ShipInfo{//船体的位置信息
		required	int64		ship_update_time = 1;		//船体数据更新时间(GPS时间,水上罗盘更新时间,导航设备时间,DP时间四者的最新时间)
		required	double		ship_lat = 2;				//经纬度位置
		required	double		ship_lon = 3;
		required	double		ship_speed = 4;				//船速
		required	double		ship_speed_ground = 5;		//地速
		required	double		ship_head = 6;				//航首向
		required	double		ship_pitch = 7;				//纵倾
		required	double		ship_roll = 8;				//横倾
		required	double		ship_course = 9;			//航向(航迹向)
		required	double		heave = 10;					//船舶升降
		required	double		wind_dir = 11;				//风向
		required	double		wind_speed = 12;			//风速
		required	double      surge_demand = 13;			//
		required	double		surge_feedback = 14;		//
		required	double		sway_demand = 15;			//
		required	double		sway_feedback = 16;			//
		required	int32		ship_mode = 17;				//船舶数据的获取方式 1:GPS; 2:DP;3:导航设备
	}
	message		PlowInfo{//犁设备信息
		required	int64		plow_update_time = 1;		//犁设备数据的更新时间(USBL更新时间，水下罗盘数据的更新时间，水下4017的更新时间)
		required	double		plow_lat = 2;				//犁设备的经纬度
		required	double		plow_lon = 3;
		required	int32		plow_pos_mode = 4;			//犁设备位置的获取方式。0:未知；1：USBL; 2:程序计算
		required	bool		plow_flag = 5;				//犁设备是否正确定位
		required	double		plow_head = 6;				//犁设备拖体首向
		required	double		plow_roll = 7;				//犁设备拖体横倾
		required	double		plow_pitch = 8;				//犁设备拖体纵倾
		required	double		plow_water_depth = 9;		//犁设备的水深
		required	double		plow_buried_depth = 10;		//犁设备的埋深
		required	double		plow_left_pump = 11;		//犁设备的左泵压力
		required	double		plow_right_pump = 12;		//犁设备的右泵压力
		required	double		plow_speed = 13;			//犁设备的速度
		required	double		plow_boots_angle = 14;		//犁设备靴角的角度
		required	double		plow_boots_length = 15;		//犁设备靴体长度
		required	double		plow_tow_length = 16;		//犁设备对应的拖缆的长度
		required	double		plow_tow_tension = 17;		//犁设备对应的拖缆的张力
		required	int64		plow_pos_update_time = 18;	//计算犁设备的速度
		required	double		plow_kp_dis = 19;			//犁设备相对于船舶的距离
	}
	message		LayInfo{//布缆的数据信息
		required	int64		cable_update_time = 1;				//布缆数据的更新时间(计米器的更新时间，水上4017的更新时间，布缆设备的更新时间)
		required	double		cable_payout_speed = 2;				//海缆缆速度
		required	double		meter_counter_coeff = 3;			//计米器系数
		required	double		meter_counter_initlen = 4;			//计米器初始读数
		required	double		meter_counter_length = 5;			//计米器读数
		required	double		cable_length = 6;					//计米器算出的海缆长度
		required	double		glj_cb_tension = 7;					//鼓轮机海缆张力
		required	double		glj_cb_speed = 8;   				//鼓轮机海缆速度
		required	double		glj_cb_length = 9;     				//鼓轮机海缆长度绝对值
		required 	double		ltj_cb_tension = 10;      			//轮胎机海缆张力
		required 	double		ltj_cb_speed = 11;   				//轮胎机海缆速度
		required 	double		ltj_cb_length = 12;     			//轮胎机海缆长度绝对值
		required 	double		tl_tension = 13;          			//拖缆张力
		required 	double		tl_length = 14;         			//拖缆长度绝对值
		required 	double		qd_tension = 15;          			//脐带张力KN
		required 	double		qd_length = 16;         			//脐带长度绝对值		
		required	int32		light_status = 17;					//灯状态
		required	int64		meter_counter_time = 18;			//计米器时间，计算速度用
		required	int32		meter_source = 19;					//海缆长度计数来源 0：未知;1:计米器;2:海缆布缆综合设备
		
	}
	message		Surface4017Data{
		required	double	surface_4017_pull1 = 1;			//水上4017拉力1
		required	double	surface_4017_pull2 = 2;			//水上4017拉力2
		required	double	surface_4017_pull3 = 3;			//水上4017拉力3
		required	double	surface_4017_lpump = 4;			//水上4017左泵压力
		required	double	surface_4017_rpump = 5;			//水上4017右泵压力
		required	int64	surface_4017_time = 6;				//水上4017更新时间
	}
	
	message		UnderWater4017Data{
		required	double	under_4017_pull1 = 1;			//水下4017拉力1
		required	double	under_4017_pull2 = 2;			//水下4017拉力2
		required	double	under_4017_pull3 = 3;			//水下4017拉力3
		required	double	under_4017_lpump = 4;			//水下4017左泵压力
		required	double	under_4017_rpump = 5;			//水下4017右泵压力 
		required	double	under_4017_water_depth = 6;					//水深(Ad4017_depth)
		required	double	under_4017_boots_angle = 7;					//靴角
		required	double	under_4017_boots_length = 8;					//靴体长度
		required	double	under_4017_buried_depth = 9;					//埋深(靴体长度) * cos(靴角);
		required	double	under_4017_touch_down_p1 = 10;					//触地1
		required	double	under_4017_touch_down_p2 = 11;					//触地2
		required	double	under_4017_touch_down_p3 = 12;					//触地3
		required	double	under_4017_touch_down_p4 = 13;					//触地4
		required	int64	under_4017_time = 14;							//水下40171更新时间
	}
	
	message		ConstructionInfo{
		required	bool	isContructed = 1;				//是否正在施工
		required	string	host_name = 2;
		required	int32	project_id = 3;
		required	int32	route_id = 4;
		required	int32	plan_id = 5;
	}
	
		
		
	message DevInfo { //有关采集设备的信息
		required	ShipInfo	ship_info = 1;
		required	PlowInfo	plow_info = 2;
		required	LayInfo		lay_info = 3;
		required	Surface4017Data	surface_4017 = 4;
		required	UnderWater4017Data under_4017 = 5;
		required	bool		surface_hmr3000_checked_flg = 6;
		required	bool		under_hmr3000_checked_flg = 7;
		required	bool		gsp_checked_flg = 8;
		required	bool		usbl_checked_flg = 9;
		required	bool		surface_4017_checked_flg = 10;
		required	bool		under_4017_checked_flg = 11;
		required	bool		metercounter_checked_flg = 12;
		required	bool		navi_device_checked_flg = 13;
		required	bool		cable_integrated_checked_flg = 14;
		required	bool		dp_checked_flg = 15;
		required	ConstructionInfo	construct_info = 16;
	}
	
	//DP设备相关的数据结构	
	//PMASNS
	message PMASNS {
		required	string		date = 1;		//日期
		required	string		time = 2;		//时间
		required	int32			gyro1_sts = 3;
		required	double		heading1 = 4;
		required	int32			gyro2_sts = 5;
		required	double		heading2 = 6;
		required	int32			gyro3_sts = 7;
		required	double		heading3 = 8;
		required	int32			mru1_sts = 9;
		required	double		pitch1 = 10;
		required	int32			roll1 = 11;
		required	double		heave1 = 12;
		required	int32			mru2_sts = 13;
		required	double		pitch2 = 14;
		required	int32			roll2 = 15;
		required	double		heave2 = 16;
		required	int32			wind1_sts = 17;
		required	double		speed1 = 18;
		required	int32			speed_dir1 = 19;
		required	double		wind2_sts = 20;
		required	int32			speed2 = 21;
		required	double		speed_dir2 = 22;		
	}
	
	//PMAGGA
	message PMAGGA {
		required	int32		gps_identifier = 1;		//日期
		required	string		time = 2;		//UTC hhmmss.ss
		required	double			lat = 3;
		required	string		lat_dir = 4;
		required	double			lon = 5;
		required	string		lon_dir = 6;
		required	int32			gps_qualifier = 7;
		required	int32		satellites_number = 8;
		required	double			hdop = 9;
		required	string		reserve1 = 10;
		required	string		reserve2 = 11;
		required	string		reserve3 = 12;
		required	string			geo_unit = 13;
		required	string		reserve4 = 14;
		required	string		reserve5 = 15;
	}
	
	//PMAVTG地面速度信息
	message PMAVTG {
		required	int32		gps_identifier = 1;		//日期
		required	double		true_course = 2;		//UTC hhmmss.ss
		required	double		magn_course = 3;
		required	double		speed_knots = 4;
		required	double		speed_kmph = 5;
	}
	//PMATHA
	message PMATHA {
		required	double		surge_demand = 1;		//日期
		required	double		surge_feedback = 2;		//UTC hhmmss.ss
		required	double		sway_demand = 3;
		required	double		sway_feedback = 4;
		required	double		yaw_demand = 5;
		required	double		yaw_feedback = 6;
	}
	
	message PMATHF {
		required	int32		thruster_number = 1;		//日期
		required	string		thruster_Status = 2;		//UTC hhmmss.ss
		required	double		rpm_pitch_demand = 3;
		required	double		rpm_pitch_feedback = 4;
		required	double		thrust_demand = 5;
		required	double		thrust_feedback = 6;
		required	double		azimuth_demand = 7;
		required	double		azimuth_feedback = 8;
	}
	
	message PMAINF {
		required	int32		refsys = 1;
		required	int32		gyro = 2;
		required	int32		mru = 3;
		required	int32		wind = 4;
		required	double		estimate_lat = 5;
		required	string		estimate_lat_dir = 6;
		required	double		estimate_lon = 7;
		required	string		estimate_lon_dir = 8;
		required	double		demand_lat = 9;
		required	string		demand_lat_dir = 10;
		required	double		demand_lon = 11;
		required	string		demand_lon_dir = 12;
		required	double		estimate_heading = 13;
		required	double		demand_heading = 14;
		required	int32		controller_gain = 15;	
	
	}
	
	message	PMAALA {
		required	int32		alarm_status = 1;
		required	double		yaw_alarm_limit = 2;
		required	bool		yaw_alarm_onoff = 3;
		required	double		surge_alarm_limit = 4;
		required	bool		surge_alarm_onoff = 5;
		required	double		sway_alarm_limit = 6;
		required	bool		sway_alarm_onoff = 7;
		required	double		xtrack_alarm_limit = 8;
		required	bool		xtrack_alarm_onoff = 9;
		
	}
	
	message	PMAVDR {
		required	int32		active_mode = 1;
		required	double		azimuth_steering_limit = 2;
		required	int32		axis_active = 3;
		required	double		turn_rate = 4;
		required	int32		steering_mode = 5;
		required	int32		steering_gain= 6;		
	}
	
	message DPPMAData {
		required	bool		gyro_sts = 1;
		required	bool		mru_sts = 2;
		required	bool		wind_sts = 3;
		required	double		heading = 4;
		required	double		course	= 5;	//cog  对地航向
		required	double		sos = 6;		//speed of ship		//船速
		required	double		sog = 7;		//speed of ground	//对地船速
		required	double		pitch = 8;		//
		required	double      roll = 9;
		required	double		heave = 10;
		required	double		water_depth = 11;
		required	double		wind_dir = 12;
		required	double		wind_speed = 13;
		required	double      surge_demand = 14;
		required	double		surge_feedback = 15;
		required	double		sway_demand = 16;
		required	double		sway_feedback = 17;
		required	double		lon	= 18;
		required	string		lon_dir = 19;
		required	double		lat = 20;
		required	string		lat_dir = 21;
	}
    